R/stan_module.R

```diff
@@ -117,6 +124,16 @@ setMethod(
     }
 )
 
+#' model_prep. Populates the model section of a StanModule
+#' @param x A StanModule object
+#' @export
+model_prep <- function(x) {
+    if (length(x@priors) > 0) x@priors <- as.list(paste(names(x@priors), "~", x@priors))
+    x@model <- paste(h_bracket(x@priors), x@model)
```

Comment:
Sorry am a bit lost here,  is this the right place to be calling `h_bracket()`  my understanding is that `h_bracket()` wraps the object in curly brackets `{  priors }`  but then you  are concatenating the model code on afterwards....  Shouldn't the model code and the priors be within the same set of curly brackets i.e.  `{ priors \n model }`

Reply:
Correct. This is my motivation for making the function generic. When dealing with the priors, the function just creates a character text given the prior list but it does not add curly brackets. When dealing with characters, it adds the curly brackets.

Reply:
Ok yer I am not keen on this design of the functions.  The idea of generics is that you abstract away implementation details but that you should get consistent or similar outcomes.  Having the method  collapse if a list or  add brackets if a string are quite different outcomes and that can lead to confusion other the purpose of the method.

Reply:
I changed the format from the S4 generic to simple function in the last commit.